src/Workspaces/CSharp/Portable/Extensions/SimpleNameSyntaxExtensions.cs

```diff
@@ -15,7 +15,15 @@ public static ExpressionSyntax GetLeftSideOfDot(this SimpleNameSyntax name)
             Contract.Requires(name.IsMemberAccessExpressionName() || name.IsRightSideOfQualifiedName() || name.IsParentKind(SyntaxKind.NameMemberCref));
             if (name.IsMemberAccessExpressionName())
             {
-                return ((MemberAccessExpressionSyntax)name.Parent).Expression;
+                var conditionalAccess = name.Parent.Parent as ConditionalAccessExpressionSyntax;
+                if (conditionalAccess != null)
+                {
+                    return conditionalAccess.Expression;
+                }
+                else
+                {
+                    return ((MemberAccessExpressionSyntax)name.Parent).Expression;
+                }
```

Comment:
Use SyntaxNodeExtensions.GetParentConditionalAccessExpression. It always finds the right node, even with more complicated conditional access.